patches/server/0937-Player-Entity-Tracking-Events.patch

```diff
@@ -4,31 +4,25 @@ Date: Wed, 30 Mar 2022 18:16:52 +0200
 Subject: [PATCH] Player Entity Tracking Events
 
 
-diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
-index 407d954935a5213a31329c4008f262c450118422..7f94da8059147760cbdc2476d0e8beda4a105f40 100644
---- a/src/main/java/net/minecraft/world/entity/Entity.java
-+++ b/src/main/java/net/minecraft/world/entity/Entity.java
-@@ -3840,9 +3840,21 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
-         EnchantmentHelper.doPostDamageEffects(attacker, target);
-     }
+diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
+index 2212f9f48636357265d8e44aba415ea4f09f1fe7..2b8b9bd78ed689cbda6ed411e948bd9d1769e12f 100644
+--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
++++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
+@@ -1611,9 +1611,16 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
+                 // CraftBukkit end
+                 if (flag) {
+                     if (this.seenBy.add(player.connection)) {
+-                        this.serverEntity.addPairing(player);
++                        // Paper start
++                        if (io.papermc.paper.event.player.PlayerTrackEntityEvent.getHandlerList().getRegisteredListeners().length == 0 || new io.papermc.paper.event.player.PlayerTrackEntityEvent(player.getBukkitEntity(), this.entity.getBukkitEntity()).callEvent()) {
++                            this.serverEntity.addPairing(player);
```

Comment:
Remove indentation change here

Reply:
Don't get what you mean. It's intended to be indented, cause it's in the condition.
Or do you prefer no indention at all to reduce the diff to 4 lines instead of 5?

Reply:
We generally prefer not to indent existing code because it creates extra diff to confict needlessly, in which not having the indent will often allow git to 3way it automatically for the most part, or at the very least often present a more useful from glancing conflict